### ISSUE TITLE
unable to call requestSubscription without storekit in ios 

### ISSUE THREAD

**Issue by jogi-gaurang** [hyochan/react-native-iap#2073]
**Description**
if i am using storekit while app runing then it is working and showing subscription popup by apple or ios but when i am not using storekit that popup is not showing

**Expected Behavior**

open apple subscription popup

**Environment:**

- react-native-iap: 8.2.2
- react-native: 0.64.3
- Platforms (iOS, Android, emulator, simulator, device): simulator

**Comment by Prathameshlabde** [comment]
https://github.com/dooboolab-community/react-native-iap/issues/2455 Can anyone please suggest on this issue Please ?

**Comment by jogi-gaurang** [comment]
create public storekit certificates

**Comment by ch3tan03** [comment]
> create public storekit certificates

how to validate the receipts locally using a certificate? your help would be appreciated

**Comment by jogi-gaurang** [comment]
you can use this function for validate ios purchases receipt 

`

validate(resData){
		if(resData){
			const ReceiptData = {
				"receipt-data": resData,
				"password": this.iosHelper.IOSSECRAT(),
			}
			const result = validateReceiptIos(ReceiptData, true).catch((e) => {
				console.log(e.code, e.message);
			}).then((returnData) => {
				try{
					console.log(returnData)
				} catch(e){
					console.log(e.code)
				}
			})
		} else {
			console.log('No Receipt Data Is There.');
		}
	}

`

**Comment by ch3tan03** [comment]
> validate(resData){ if(resData){ const ReceiptData = { "receipt-data": resData, "password": this.iosHelper.IOSSECRAT(), } const result = validateReceiptIos(ReceiptData, true).catch((e) => { console.log(e.code, e.message); }).then((returnData) => { try{ console.log(returnData) } catch(e){ console.log(e.code) } }) } else { console.log('No Receipt Data Is There.'); } }

thanks for the reply. I'm also doing the same but getting below response code in error 
{status: 21002}

**Comment by jogi-gaurang** [comment]
yes its because of you are using storekit you have to create public storekit certificate and run your app in real device

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
